### PR TITLE
Refacto Repository tests

### PR DESCRIPTION
**Pull request by Wauplin**
TL;DR:
1. Sorry for huge diff :cry: 
2. Repository tests work on Windows (related to https://github.com/huggingface/huggingface_hub/pull/1112)
3. Repository tests run in parallel
4. Repository tests use `/tmp` dir
5. Less redundancy

----
First of all, sorry for the huge diff in the PR :cry: .
I started it as a preliminary work for https://github.com/huggingface/huggingface_hub/pull/1112 (Windows CI):

1. I started to refacto how paths are generated => no more `"directory/whatever_file.txt"` as it breaks on Windows
6. While doing that, I removed the `WORKING_DIR_FIXTURE` to replace it by a proper `/tmp` folder
=> Avoid to have to clean it after tests (when tests were failing, I often ended up with untracked folders in my `huggingface_hub` folder that I had to delete)
7. Then I read more the tests and realized we were doing _a lot_ of redundant work (creating 1 new repo on the Hub for each test even though we do not modify it)
    1. I've split repository tests between `TestRepositoryShared` and `TestRepositoryUniqueRepos`. In shared test, a single repo is created on the Hub and cloned multiple times. No files are pushed to this repo during the tests.
    => Save quite some time of setup/teardown. In Unique test, a repo is created per test (same as before) so that they are independent from each other.
    1. I also added attributes like `self.repo_id`, `self.repo_path` and `self.repo_url` and an helper `self.clone_from(...)` to avoid reduncancy (e.g. recomputing `repo_id=f"{USER}/{self.REPO_NAME}", local_dir="{WORKING_DIR}/{REPO_NAME}"` all the time).
8. And finally I removed some tests that were duplicated (*).

=> Now we have a `Repository` test suite that should be "iso" compared to before but running in parallel (~1min instead of 8) **and on Windows**.

(*) For example [in dataset tests](https://github.com/huggingface/huggingface_hub/pull/1277/files#diff-ed4fdd46616101997cc333941694ea7e53f822b82fd75cf53dfa586f902ddfdaL1621), we are testing that cloning from a different repo type works. But also testing that we can commit, pull, push,... which is quite redundant because what we really want is to check if a repo can be cloned with a repo_type. All subsequent actions are already tested separately ("test only 1 feature at a time").

(**EDIT:** "non repository" tests are failing but it's not due to this PR :confused:)

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Much cleaner this way! Thanks for the refactor @Wauplin.

**Comment by Wauplin** [comment]
Thanks @LysandreJik for the review and sorry again for the big diff to review. I'll fix conflicts, wait for the checks and merge it :)
